Lib/test/test_unparse.py

```diff
@@ -130,7 +130,23 @@ class Foo: pass
 
 class ASTTestCase(unittest.TestCase):
     def assertASTEqual(self, ast1, ast2):
-        self.assertEqual(ast.dump(ast1), ast.dump(ast2))
+        missing = object()
+        def compare(a, b):
```

Comment:
Defining a function at each assertASTEqual() call sounds expensive. Can you pass self parameter to compare() and move it as the class level (method or staticmethod), or even at the module level (only define it once)?

For example:

```
     def _assertASTEqual(self, a, b):  # accept AST nodes, lists and objects
         ... # your code

     def assertASTEqual(self, ast1, ast2):  # accept AST nodes
         self._assertASTEqual(ast1, ast2)
```

Reply:
I timed three approaches to see exactly the costs involved.  All test runs were performed using:
  `amd64\python_d.exe -m test -uall test_unparse`
All three versions, inner function, module function and method come in at 59s +/- 0.2s over 5 runs.  Contrasted to the current runtime of 95s.

My chosen approach is modeled loosely after `ast.dump()` by using the inner function.  An advantage of the inner function is that the `missing` value does not need to pollute the function signature nor global namespace.  Ultimately, the design is just stylistic.

Reply:
If you prefer to redefine the function at each call, would you mind to add a comment explaining why you prefer that?

My worry is mostly about performance. If there is no impact on performance, you can leave it as it is.

Reply:
I do note, however, in micro-benchmarks *creating* inner functions have a substantial runtime cost.  But in this case, that time is dwarfed by time spent *calling* that function.